File: Furina_Betting_Games.py
from random import randrange
balance = 500

while(balance > 0):
    print(f"Welcome to the Furina Betting Games. Your current balance is {balance} Furidollars.")
    print("0: Double or Nothing Coin flip (You're heads).")
    gameChoice = input("1: Dice Difference game with adjustable betting. Pick a game: ")

    #Exception handling
    match gameChoice:
        case "0": #Game 1's Logic
            ran = randrange(2)
            #Set balance and display text in the case that 'ran' is 0
            balance *= 2 if ran == 0 else 0
            coin = "Heads" if ran == 0 else "Tails"
            resultText = "You won! Your balance is doubled! You may now rejoice in this~\n" if ran == 0 else "You lost. What's done is done~.\n"
            print(f"Flipping your coin... It landed on {coin}!", resultText)
        case "1": #Game 2's Logic
            betAmount, dice1, dice2 = 0, randrange(6) + 1, randrange(6) + 1 #Generate a random integer between 1 and 6 for both dice
            diff = abs(dice1 - dice2)

            #Exception handling
            while (betAmount > balance or betAmount < 0):
                try:
                    betAmount = int(input("How much do you want to bet: "))
                except:
                    print("A negative bet? Did you hit your head or...?\n") if betAmount < 0 else print("You don't even have that much, hmph! Try again!\n\n")

            betChoice = input("0: Lower than 3\n1: Higher than 2\nHurry up and pick a number: ")
            #Exception handling
            match betChoice:
                case "0": betChoice = 0
                case "1": betChoice = 1
                case _:    
                    print("You can only enter 0 or 1, fufu~ Try again!\n")
                    pass

            print(f"Your dice rolled: {dice1} and {dice2}. Their difference is {diff}")
            if ((betChoice == 0 and diff < 3) or (betChoice == 1 and diff > 2)):
                balance += betAmount
                print("I can't believe you won... ugh... good job!\n")
            else:
                balance -= betAmount
                print("Wueheheh, you lost again!\n")    
        case _: 
            print("That isn't an available game, c\'mon!\n")
            pass

print("Thank you for playing! I will now invest this money into Fontinalia Mousses~")

File: Furina_Betting_Games.js
//Apparently, getting user input with prompt() doesn't work in VSCode, so this code only works in a browser
//Otherwise, it is almost identical to C# in terms of syntax, with the exclusion of types. I wonder what Typescript is like...
function handleGameCoinflip(balance) {
    let rand = Math.floor(Math.random() * 2); //Generate a random integer between 0 and 1
    
    let coin = (rand === 0) ? "Heads" : "Tails";
    let finisher = (rand === 0) ? "You won! Your balance is doubled! You may now rejoice in this!" : "You lost. What's done is done~.";
    console.log("Flipping your coin... It landed on " + coin + "!\n" + finisher + "\n");
    return balance = (rand == 0) ? balance * 2 : 0;
}

function handleGameDice(balance) {
    //Generates a random integer between 1 and 6 for both dice, then get their difference
    let dice1 = Math.floor(Math.random() * 6) + 1, dice2 = Math.floor(Math.random() * 6) + 1, diff = Math.abs(dice1 - dice2); 
    
    let betAmount = 0
    //Exception Handling
    try { betAmount = parseInt(prompt("How much do you want to bet: ")); }
    catch (err) { console.log("Stop messing with me!\n"); return balance; }
    if (betAmount > balance) { console.log("You don't even have that much, hmph! Try again!\n"); return balance; }
    if (betAmount < 0) { console.log("A negative bet? Did you hit your head or...?\n"); return balance; }
    
    let betChoice = 0;
    let input = prompt("\n0: Lower than 3\n1: Higher than 2\nHurry up and pick a number: ");
    switch (input) { //Exception handling for betChoice
            case "0": betChoice = 0; break;
            case "1": betChoice = 1; break;
            default: console.log("You can only enter 0 or 1, fufu~ Try again!\n"); return balance;
    }

    //Game 2's Logic
    console.log("Your dice rolled: " + dice1 + " and " + dice2 + ". Their difference is " + diff);
    if ((betChoice == 0 && diff < 3) || (betChoice == 1 && diff > 2)) {
        balance += betAmount;
        console.log("I can't believe you won... ugh... good job!\n\n");
    }
    else {
        balance -= betAmount;
        console.log("Wueheheh, you lost again!\n\n");
    }
    return balance;    
}

function mainGameLoop(balance) {
    while (balance > 0) {
        console.log("Welcome to the Furina Betting Games. Your current balance is: " + balance + " Furidollars.");
        console.log("\n0: Double or Nothing Coin flip (You're heads).");
        let gameChoice = prompt("\n1: Dice Difference game with adjustable betting. Pick a game: ");
        switch (gameChoice) {
                case "0": balance = handleGameCoinflip(balance); break;
                case "1": balance = handleGameDice(balance); break;
                default: console.log("That isn't an available game, c\'mon!\n"); continue;
        }
    }
}

mainGameLoop(500);
console.log("Thank you for playing! I will now invest this money into Fontinalia Mousses~");